src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -144,15 +143,34 @@ public InternalEngine(EngineConfig engineConfig, boolean skipInitialTranslogReco
             this.mergeSchedulerListener = new MergeSchedulerListener();
             this.mergeScheduler.addListener(mergeSchedulerListener);
             this.mergeScheduler.addFailureListener(mergeSchedulerFailureListener);
-            final TranslogRecoveryPerformer transformer = engineConfig.getTranslogRecoveryPerformer();
             try {
-                long nextTranslogID = translogId.v2();
-                translog.newTranslog(nextTranslogID);
-                translogIdGenerator.set(nextTranslogID);
-                if (translogId.v1() != null && skipInitialTranslogRecovery == false) {
-                    recoverFromTranslog(translogId.v1(), transformer);
+                if (skipInitialTranslogRecovery) {
+                    // make sure we point at the latest translog from now on..
+                    commitIndexWriter(writer, translog.currentId());
                 } else {
-                    flush(true, true);
+                    if (committedTranslogId != null) {
+                        try {
+                            // trim unneeded files
+                            translog.committedTranslogId(committedTranslogId);
+                        } catch (FileNotFoundException ex) {
```

Comment:
I explicitly throw FileNotFoundException, similar to the old code. I can't add it because it isn't thrown.